app/models/work_statistic.rb

```diff
@@ -76,59 +75,49 @@ def pct_translation_completed
-      self[:translated_pages] = work.pages.where("translation_status = '#{Page::STATUS_TRANSLATED}'").count
-      self[:blank_pages] = work.pages.where("status = '#{Page::STATUS_BLANK}'").count
-      self[:translated_blank] = work.pages.where("translation_status = '#{Page::STATUS_BLANK}'").count
-      self[:annotated_pages] = work.pages.where("status = '#{Page::STATUS_INDEXED}'").count
-      self[:translated_annotated] = work.pages.where("translation_status = '#{Page::STATUS_INDEXED}'").count
-      self[:needs_review] = work.pages.where("status = '#{Page::STATUS_NEEDS_REVIEW}'").count
-      self[:translated_review] = work.pages.where("translation_status = '#{Page::STATUS_NEEDS_REVIEW}'").count
-    end
-    self[:complete] = self.pct_completed
-    self[:translation_complete] = self.pct_translation_completed
-    save!
+  def recalculate(_options = {})
+    stats = get_stats_hash
+
+    recalculate_from_hash(stats)
 
     recalculate_parent_statistics
   end
 
+  def recalculate_from_hash(stats = {})
+    self[:total_pages] = stats[:total]
 
+    self[:transcribed_pages]  = stats[:transcription][Page::STATUS_TRANSCRIBED] || 0
+    self[:corrected_pages]    = stats[:transcription][Page::STATUS_TRANSCRIBED] || 0
+    self[:blank_pages]        = stats[:transcription][Page::STATUS_BLANK] || 0
+    self[:annotated_pages]    = stats[:transcription][Page::STATUS_INDEXED] || 0
+    self[:needs_review]       = stats[:transcription][Page::STATUS_NEEDS_REVIEW] || 0
 
-private
+    self[:translated_pages]     = stats[:translation][Page::STATUS_TRANSLATED] || 0
+    self[:translated_blank]     = stats[:translation][Page::STATUS_BLANK] || 0
+    self[:translated_annotated] = stats[:translation][Page::STATUS_INDEXED] || 0
+    self[:translated_review]    = stats[:translation][Page::STATUS_NEEDS_REVIEW] || 0
 
-  # current logic to recalculate statistics for parent document set and parent collection
-  def recalculate_parent_statistics
-    #save completed information for collections/document sets
-    self.work.collection.calculate_complete unless self.work.collection.nil?
-    if !self.work.document_sets.empty?
-      self.work.document_sets.each do |set|
-        set.calculate_complete
-      end
-    end    
+    self[:complete]             = pct_completed
+    self[:translation_complete] = pct_translation_completed
+    
+    save!
   end
 
+  def get_stats_hash
+    {
+      transcription: work.pages.group(:status).count,
+      translation: work.pages.group(:translation_status).count,
+      total: work.pages.count
+    }
+  end
+  private
 
+  # current logic to recalculate statistics for parent document set and parent collection
+  def recalculate_parent_statistics
+    # save completed information for collections/document sets
+    work.collection&.calculate_complete
+    unless work.document_sets.empty?
+      work.document_sets.each(&:calculate_complete)
```

Comment:
And that `&:`, too?

Reply:
Here it's being used as a stand in for `.each(|d| d.calculate_complete)`. Just a shorthand.